mobile/test/swift/integration/proxying/HTTPRequestUsingPacProxyTest.swift

```diff
@@ -0,0 +1,92 @@
+import Envoy
+import EnvoyEngine
+import EnvoyTestApi
+import EnvoyTestServer
+import Foundation
+import TestExtensions
+import XCTest
+
+final class HTTPRequestUsingPacProxyTest: XCTestCase {
+  override static func setUp() {
+    super.setUp()
+    register_test_extensions()
+  }
+
+  override static func tearDown() {
+    super.tearDown()
+    // Flush the stdout and stderror to show the print output.
+    fflush(stdout)
+    fflush(stderr)
+  }
+
+  private func executeRequest(engine: Engine, scheme: String, authority: String) -> String? {
+    let responseHeadersExpectation =
+        self.expectation(description: "Successful response headers received")
+    let responseTrailersExpectation =
+        self.expectation(description: "Successful response trailers received")
+
+    let requestHeaders = RequestHeadersBuilder(method: .get, scheme: scheme,
+                                               authority: authority, path: "/")
+      .build()
+
+    var responseBuffer = Data()
+    engine.streamClient()
+      .newStreamPrototype()
+      .setOnResponseHeaders { responseHeaders, _, _ in
+         XCTAssertEqual(200, responseHeaders.httpStatus)
+         responseHeadersExpectation.fulfill()
+      }
+      .setOnResponseData { data, _, _ in
+        responseBuffer.append(contentsOf: data)
+      }
+      .setOnResponseTrailers { _, _ in
+        responseTrailersExpectation.fulfill()
+      }
+      .start()
+      .sendHeaders(requestHeaders, endStream: true)
+
+    let expectations = [responseHeadersExpectation, responseTrailersExpectation]
+    XCTAssertEqual(XCTWaiter.wait(for: expectations, timeout: 10), .completed)
+
+    return String(data: responseBuffer, encoding: .utf8)
+  }
+
+  func testHTTPRequestUsingPacProxy() throws {
+    EnvoyTestServer.startHttpProxyServer()
+    let proxyPort = EnvoyTestServer.getProxyPort()
+    EnvoyTestServer.startHttp1PlaintextServer()
+    let httpPort = EnvoyTestServer.getHttpPort()
+
+    let engineExpectation = self.expectation(description: "Run started engine")
+
+    let engine = EngineBuilder()
+      .setLogLevel(.debug)
+      .setLogger { _, msg in
+        print(msg, terminator: "")
+      }
+      .setOnEngineRunning {
+        engineExpectation.fulfill()
+      }
+      .respectSystemProxySettings(true)
+      .build()
+
+    let pacScript = """
+    function FindProxyForURL(url, host) {
+      return "PROXY 127.0.0.1:\(proxyPort)";
+    }
+    """;
+    EnvoyTestServer.setHeadersAndData("Content-Type", header_value: "application/x-ns-proxy-autoconfig", response_body: pacScript)
+    EnvoyTestApi.registerTestProxyResolver("127.0.0.1", port: httpPort, usePacResolver: true)
```

Comment:
I don't understand where this test tells the stack to use a PAC file. I went looking through the source for `EnvoyTestApi` but didn't find it there either. For this to work you're going to have to tell the stack to use a certain PAC file, and then you need something to serve that PAC file

Reply:
This part ensures that the `proxy_autoconfiguration_file_url` here: https://github.com/envoyproxy/envoy/pull/36477/files#diff-3c1c391c9cebec316f969380f3858fccec0afbcac44ffdc04c9a3097a24e1ee7R97 is set to `127.0.0.1:{httpPort}`.  I've printed the `cf_proxy_autoconfiguration_file_url` here https://github.com/envoyproxy/envoy/pull/36477/files#diff-3c1c391c9cebec316f969380f3858fccec0afbcac44ffdc04c9a3097a24e1ee7R89 that is passed to `CFNetworkExecuteProxyAutoConfigurationURL` and the value is indeed `127.0.0.1:{httpPort}`.  So the `CFNetworkExecuteProxyAutoConfigurationURL` API should be trying to fetch the PAC script from `127.0.0.1:{httpPort}`.